File: triatlon-cs/Repository/RaceRepository.cs
using triatlon_cs.Model.Activity;

namespace triatlon_cs.Repository;

// RaceRepository

public abstract class RaceRepository : IRepository<int, AthletePoints>
{
    
    // Properties

    private RaceType _raceType;

    public RaceType RaceType
    {
        get => _raceType;
    }
    
    // Lifecycle

    protected RaceRepository(RaceType raceType)
    {
        _raceType = raceType;
    }

    protected RaceRepository(Race race)
    {
        _raceType = race.Type;
    }
    
    // IRepository Methods
    
    public abstract AthletePoints FindOne(int id);
    public abstract IEnumerable<AthletePoints> FindAll();
    public abstract AthletePoints Save(int id);
    public abstract AthletePoints Delete(int id);
    public abstract AthletePoints Update(AthletePoints entity);
    
}

File: triatlon-java/src/main/java/repository/RaceRepository.java
package repository;

import model.activity.AthletePoints;
import model.activity.Race;
import model.activity.RaceType;

// RaceRepository

public abstract class RaceRepository implements Repository<Integer, AthletePoints> {

    // Private Properties

    private RaceType raceType;

    // Lifecycle

    public RaceRepository(RaceType raceType) {
        this.raceType = raceType;
    }

    public RaceRepository(Race race) {
        this.raceType = race.getType();
    }

    // Getters & Setters

    public RaceType getRaceType() {
        return this.raceType;
    }

    public void setRaceType(RaceType raceType) {
        this.raceType = raceType;
    }

}
